python/GafferUI/SpreadsheetUI.py

```diff
@@ -1212,6 +1315,276 @@ def __fixedWidth( cls, plugValueWidget ) :
+			confirmLabel = "Rename",
+		).waitForText( parentWindow = self.ancestor( GafferUI.Window ) )
+
+		if not newSectionName or newSectionName == sectionName :
+			return
+
+		if newSectionName in sectionNames :
+			for suffix in range( 1, len( sectionNames ) + 2 ) :
+				suffixedSectionName = newSectionName + str( suffix )
+				if suffixedSectionName not in sectionNames :
+					newSectionName = suffixedSectionName
+					break
+
+		with Gaffer.UndoScope( self.__rowsPlug.ancestor( Gaffer.ScriptNode ) ) :
+
+			# Move appropriate columns to renamed section
+			for cellPlug in self.__rowsPlug["default"]["cells"] :
+				m = Gaffer.Metadata.value( cellPlug, "spreadsheet:section" )
+				if m == sectionName or ( not m and sectionName == "Other" ) :
+					Gaffer.Metadata.registerValue( cellPlug, "spreadsheet:section", newSectionName )
+
+			# Reapply section order for renamed section.
+			self.__assignSectionOrder(
+				self.__rowsPlug,
+				[ newSectionName if n == sectionName else n for n in sectionNames ]
+			)
+
+			# And choose the renamed tab if necessary.
+			if sectionIsCurrent :
+				self._qtWidget().setCurrentIndex( sectionNames.index( sectionName ) )
+
+	def __deleteSection( self, sectionName ) :
+
+		sectionNames = self.sectionNames( self.__rowsPlug )
+
+		with Gaffer.UndoScope( self.__rowsPlug.ancestor( Gaffer.ScriptNode ) ) :
+			# Iterate in reverse to avoid invalidating column indices we're
+			# about to visit.
+			for columnIndex, cellPlug in reversed( list( enumerate( self.__rowsPlug["default"]["cells"] ) ) ) :
+				m = Gaffer.Metadata.value( cellPlug, "spreadsheet:section" )
+				if m == sectionName or ( not m and sectionName == "Other" ) :
+					self.__rowsPlug.removeColumn( columnIndex )
+			# Reassign section indices to remove gap.
+			self.__assignSectionOrder(
+				self.__rowsPlug,
+				[ n for n in sectionNames if n != sectionName ]
+			)
+
+	def __moveSection( self, fromSectionName, toSectionName ) :
+
+		sectionNames = self.sectionNames( self.__rowsPlug )
+		sectionIsCurrent = self._qtWidget().tabText( self._qtWidget().currentIndex() ) == fromSectionName
+
+		with Gaffer.UndoScope( self.__rowsPlug.ancestor( Gaffer.ScriptNode ) ) :
+
+			# Move columns
+			for cellPlug in self.__rowsPlug["default"]["cells"] :
+				m = Gaffer.Metadata.value( cellPlug, "spreadsheet:section" )
+				if m == fromSectionName or ( not m and fromSectionName == "Other" ) :
+					Gaffer.Metadata.registerValue( cellPlug, "spreadsheet:section", toSectionName )
+
+			# Reapply section order to remove gaps.
+			newSectionNames = [ n for n in sectionNames if n != fromSectionName ]
+			self.__assignSectionOrder( self.__rowsPlug, newSectionNames )
+
+			if sectionIsCurrent :
+				self._qtWidget().setCurrentIndex( newSectionNames.index( toSectionName ) )
+
+	def __contextMenuRequested( self, pos ) :
+
+		m = IECore.MenuDefinition()
+		sectionName = self._qtWidget().tabText( self._qtWidget().tabAt( pos ) )
+
+		m.append(
+			"/Rename",
```

Comment:
Re: earlier comment re the behaviour shift once you've re-named `Other`, would it make sense to dis-allow re-naming of the Other section? I know that makes a simple binary partition of plugs into two sections more complex though, but I wonder how common that case is.

Reply:
I've kept the ability to rename "Other", as it does binary partitioning easier.